http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesServiceDbQueries.scala

```diff
@@ -31,20 +38,26 @@ object KubernetesServiceDbQueries {
       labelFilter
     )
 
-  def listAppsByWorkspaceId(workspaceId: Option[WorkspaceId],
-                            labelFilter: LabelMap = Map(),
-                            includeDeleted: Boolean = false
+  /**
+   * List all apps in the given workspace, with optional label filter.
+   * This method should be used by v2 app routes.
+   */
+  def listFullAppsByWorkspaceId(workspaceId: Option[WorkspaceId],
+                                labelFilter: LabelMap = Map(),
+                                includeDeleted: Boolean = false
   )(implicit
     ec: ExecutionContext
   ): DBIO[List[KubernetesCluster]] =
     joinFullAppAndUnmarshal(
-      listClustersByWorkspaceId(workspaceId),
+      kubernetesClusterQuery,
       nodepoolQuery,
-      if (includeDeleted) appQuery else nonDeletedAppQuery,
+      filterByWorkspaceId(if (includeDeleted) appQuery else nonDeletedAppQuery, workspaceId),
       labelFilter
     )
 
-  // Called by MonitorAtBoot to determine which apps need monitoring
+  /**
+   * List all apps that need monitoring. Called by MonitorAtBoot.
```

Comment:
TODO for myself: look at `MonitorAtBoot` code, make sure it works for v2 apps

Reply:
Looking at `MonitorAtBoot` code, it looks like it explicitly doesn't handle Azure, which is good (better than trying to publish the wrong message). Thinking I make a separate ticket/PR for `MonitorAtBoot` logic.

Reply:
MonitorAtBoot: https://broadworkbench.atlassian.net/browse/TOAZ-281